subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftLibraryIntegrationTest.groovy

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift
+
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.NativeLanguageRequirement
+import org.gradle.nativeplatform.fixtures.RequiresSupportedLanguage
+import org.gradle.nativeplatform.fixtures.app.ExeWithLibraryUsingSwiftLibraryHelloWorldApp
+import org.gradle.nativeplatform.fixtures.app.SwiftHelloWorldApp
+
+import static org.gradle.util.Matchers.containsText
+
+@RequiresSupportedLanguage(NativeLanguageRequirement.SWIFT)
+class SwiftLibraryIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
+    def helloWorldApp = new SwiftHelloWorldApp()
+
+    def "build fails when compilation fails"() {
+        given:
+        buildFile << """
+            apply plugin: 'swift-library'
+         """
+
+        and:
+        helloWorldApp.brokenFile.writeToDir(file("src/main"))
+
+        expect:
+        fails "assemble"
+        failure.assertHasDescription("Execution failed for task ':compileSwift'.");
+        failure.assertHasCause("A build operation failed.")
+        failure.assertThatCause(containsText("Swift compiler failed while compiling swift file(s)"))
+    }
+
+    def "sources are compiled with Swift compiler"() {
+        settingsFile << "rootProject.name = 'hello'"
+
+        given:
+        helloWorldApp.librarySources*.writeToDir(file('src/main'))
+
+        and:
+        buildFile << """
+            apply plugin: 'swift-library'
+         """
+
+        expect:
+        succeeds "assemble"
+        result.assertTasksExecuted(":compileSwift", ":assemble")
+        sharedLibrary("build/lib/hello").assertExists()
+    }
+
+    def "can define public module"() {
+        settingsFile << "rootProject.name = 'hello'"
+        given:
+        def app = new SwiftHelloWorldApp()
+        app.library.sourceFiles.each { it.writeToFile(file("src/main/swift/$it.name")) }
+
+        and:
+        buildFile << """
+            apply plugin: 'swift-library'
+         """
+
+        expect:
+        succeeds "assemble"
+        result.assertTasksExecuted(":compileSwift", ":assemble")
+        sharedLibrary("build/lib/hello").assertExists()
+        file("build/lib/hello.swiftmodule").assertExists()
+    }
+
+    def "can compile and link against another library"() {
+        settingsFile << "include 'lib1', 'lib2'"
+        def app = new ExeWithLibraryUsingSwiftLibraryHelloWorldApp()
+
+        given:
+        buildFile << """
+            project(':lib1') {
+                apply plugin: 'swift-library'
+                dependencies {
+                    implementation project(':lib2')
+                }
+                tasks.withType(SwiftCompile)*.moduleName = 'Hello'
+            }
+            project(':lib2') {
+                apply plugin: 'swift-library'
+                tasks.withType(SwiftCompile)*.moduleName = 'Greeting'
```

Comment:
This is really a separate feature that probably should be in a separate test case and as a separate story: The feature is to allow me to declare the name of the module and verify that the output files have the correct names and the consumer can reference the module using this name. It would be a similar feature to declaring the name for library.